src/agent/ppo.py

```diff
@@ -0,0 +1,315 @@
+import gc
```

Comment:
Nathanial and Jonathan will you have a look at this implementation?
It uses a linear policy head but it shouldn't be too much work to have also an alternative that uses token generation.